### PR TITLE
[CI/CD] create build and test action

### PR DESCRIPTION
**Pull request by kroggen**

### CODE REVIEW

**Comment by kroggen** [comment]
Changes to the `master` and `develop` branches must be done via pull requests, and the above action will run and must pass before any pull request is merged

**Comment by kroggen** [comment]
There is no need to merge the `develop` into `release/2.4.4` after this is merged

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      - reopened
+      - synchronize
+  push:
+    branches:
+      - master
+      - develop
+
+jobs:
+  build_and_test:
+    runs-on: ubuntu-latest
```
Comment (hayarobi) [inline .github/workflows/build_and_test.yml:20]: Is this means that test will run on different ubuntu version depending on the testing time?

Comment (kroggen) [reply .github/workflows/build_and_test.yml:20]: It will run on the last version of ubuntu available for actions

Comment (kroggen) [reply .github/workflows/build_and_test.yml:20]: One test run passed with the current config:

https://github.com/aergoio/aergo/actions/runs/4931682161/jobs/8813966127

**Comment by kroggen** [comment]
It uses `go test -v ./...` instead of `make check` so we can see which specific test failed

**Pull request merged**